gap/hash.gi

```diff
@@ -761,7 +761,7 @@ function(p,data)
     fi;
    fi;
    # now we have a Perm2Rep:
-   return HashKeyBag(p,255,0,2*l) mod data + 1;
```

Comment:
One thing I am concerned about is that with this change, `orb` essentially becomes incompatible with GAP 4.8 and older.

I really would like to have a way to make this work correctly either way. HRM. I'll think about it a bit.

Reply:
Aha, on Slack you suggested to use `ADDR_PERM(p) - ADDR_OBJ(p)` to determine the offset on the C level. So one way to resolve this would be to check that difference in C for a test perm, and then export the diff into a GAP variable and use that here. Or else

Reply:
Thanks.

Reply:
Is there anything further that I should do to get this merged? @fingolfin